File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandSearchArticle.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandSearchArticle extends TelnetCommand {
    String _author;
    String _gy;
    String _keyword;
    String _mark;

    public BahamutCommandSearchArticle(String keyword, String author, String mark, String gy) {
        _keyword = keyword;
        _author = author;
        _mark = mark;
        _gy = gy;
        Action = SearchArticle;
    }

    public void execute(TelnetListPage aListPage) {
        TelnetClient.getClient().sendStringToServerInBackground("~" + _keyword + "\n" + _author + "\n" + _mark + "\n" + _gy);
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[SearchArticle][keyword=" + _keyword + " author=" + _author + " mark=" + _mark + " gy=" + _gy + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/command/BahamutCommandSearchArticle.kt
package com.kota.Bahamut.command

import com.kota.Bahamut.listPage.TelnetListPage
import com.kota.Bahamut.listPage.TelnetListPageBlock
import com.kota.telnet.TelnetClient

class BahamutCommandSearchArticle(
    var keyword: String,
    var author: String?,
    var mark: String?,
    var gy: String?
) : TelnetCommand() {
    init {
        action = BahamutCommandDef.Companion.SEARCH_ARTICLE
    }

    override fun execute(telnetListPage: TelnetListPage) {
        TelnetClient.client!!
            .sendStringToServerInBackground("~$keyword\n$author\n$mark\n$gy")
    }

    override fun executeFinished(telnetListPage: TelnetListPage, telnetListPageBlock: TelnetListPageBlock) {
        isDone = true
    }

    override fun toString(): String {
        return "[SearchArticle][keyword=$keyword author=$author mark=$mark gy=$gy]"
    }
}
